hooks/hookexecution/execution_test.go

```diff
@@ -0,0 +1,151 @@
+package hookexecution
+
+import (
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/hooks"
+	"github.com/prebid/prebid-server/v2/hooks/hookstage"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"github.com/prebid/prebid-server/v2/privacy"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestHandleModuleActivitiesBidderRequestPayload(t *testing.T) {
+
+	testCases := []struct {
+		description         string
+		hookCode            string
+		privacyConfig       *config.AccountPrivacy
+		inPayloadData       hookstage.BidderRequestPayload
+		expectedPayloadData hookstage.BidderRequestPayload
+	}{
+		{
+			description: "payload should change when userFPD is blocked by activity",
+			hookCode:    "foo",
+			inPayloadData: hookstage.BidderRequestPayload{
+				Request: &openrtb_ext.RequestWrapper{BidRequest: &openrtb2.BidRequest{
+					User: &openrtb2.User{ID: "test_user_id"},
+				}},
+			},
+			privacyConfig: getTransmitUFPDActivityConfig("foo", false),
+			expectedPayloadData: hookstage.BidderRequestPayload{
+				Request: &openrtb_ext.RequestWrapper{BidRequest: &openrtb2.BidRequest{
+					User: &openrtb2.User{ID: ""},
+				},
+				}},
+		},
+		{
+			description: "payload should not change when userFPD is not blocked by activity",
+			hookCode:    "foo",
+			inPayloadData: hookstage.BidderRequestPayload{
+				Request: &openrtb_ext.RequestWrapper{BidRequest: &openrtb2.BidRequest{
+					User: &openrtb2.User{ID: "test_user_id"},
+				}},
+			},
+			privacyConfig: getTransmitUFPDActivityConfig("foo", true),
+			expectedPayloadData: hookstage.BidderRequestPayload{
+				Request: &openrtb_ext.RequestWrapper{BidRequest: &openrtb2.BidRequest{
+					User: &openrtb2.User{ID: "test_user_id"},
+				}},
+			},
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.description, func(t *testing.T) {
+			activityControl := privacy.NewActivityControl(test.privacyConfig)
+			newPayload := runHandleModuleActivities(test.hookCode, activityControl, test.inPayloadData)
+			assert.Equal(t, test.expectedPayloadData.Request.BidRequest, newPayload.Request.BidRequest)
+		})
+	}
+}
+
+func TestHandleModuleActivitiesProcessedAuctionRequestPayload(t *testing.T) {
```

Comment:
It looks like processed auction request works because the payload type is able to be casted to the bidder request payload type, is that correct?
Are these the only two payload types we care about because they both deal with a bid request?

Reply:
Correct, `ProcessedAuctionRequestPayload` and `BidderRequestPayload` are only payloads with BidderRequest and they both implement `PayloadBidderRequest interface`. 
This makes them to pass type check here `payloadData, ok := any(&payload).(hookstage.PayloadBidderRequest)`